crates/gen/src/delegate32.rs

```diff
@@ -4,23 +4,45 @@ use squote::{quote, TokenStream};
 #[derive(Debug)]
 pub struct Delegate32 {
     pub name: TypeName,
+    pub method: Method,
 }
 
 impl Delegate32 {
     pub fn from_type_name(name: TypeName) -> Self {
-        Self { name }
+        let method = name
+            .def
+            .methods()
+            .find(|method| method.name() == "Invoke")
+            .unwrap();
+
+        let method = Method::from_method_def(&method, 0, &[], &name.namespace);
+        Self { name, method }
+    }
+
+    pub fn dependencies(&self) -> Vec<winmd::TypeDef> {
+        self.method.dependencies()
     }
 
     pub fn gen(&self) -> TokenStream {
-        let definition = self.name.gen_definition();
+        let name = self.name.gen();
+
+        // TODO: here we're using gen_abi() because as an ABI function pointer it can't use the owning
+        // types. Perhaps we should introduce a NonOwning<T> concept for this case.
```

Comment:
Is it possible to use `&T` instead? A `NonOwning<T>` would still need to be sure that it doesn't last longer than the `T` does which is exactly what `&T` tracks. If that's too difficult a `NonOwning<T>` would be fine, but it's use would always be unsafe since it's up to the programmer to track the lifetime.

Reply:
This is one case where I don't have an abstraction over what the developer would write. These delegates are just function pointers and unless I come up with an abstraction, we need to honor the ABI directly. `T&` won't work if `T` is `NonNull` or some type that is not blittable. The ABI rules get in the way. Hopefully I can find a way to abstract this ABI the same as others and then we can just use the regular `Param<T>` than handles all of this, but in this case there's no "object" to hang it off.